### PR TITLE
Add emoji reactions to Comments

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/3059

This is work in progress. Will push up work as I go along. :smile:

### CODE REVIEW

File: components/conversations/Comment.js [hunk 1]
```diff
@@ -314,6 +360,93 @@ const Comment = ({ comment, canEdit, canDelete, withoutActions, maxCommentHeight
             )
           }
         </InlineEditField>
+        {Object.entries(selectedReactions).map(([emoji, count]) => {
+          return (
+            <StyledLink
```
Comment (Betree) [inline components/conversations/Comment.js:365]: If it does not have an `href`, maybe we can use something else than a link for that? What about `Span`?

File: components/conversations/Comment.js [hunk 2]
```diff
@@ -314,6 +360,93 @@ const Comment = ({ comment, canEdit, canDelete, withoutActions, maxCommentHeight
...
+                          </StyledLink>
+                        );
+                      })}
+                    </Box>
+                  </Container>
+                )}
+              </Popper>
+            )}
+          </span>
+        </Manager>
```
Comment (Betree) [inline components/conversations/Comment.js:449]: Can we extract that to a separate component, like `CommentReactionPicker`?

**Comment by Betree** [comment]
@SudharakaP I'm gonna add some more commits on this PR and API's to demo the feature to the team tomorrow

**Comment by SudharakaP** [comment]
> @SudharakaP I'm gonna add some more commits on this PR and API's to demo the feature to the team tomorrow

Wonderful. Feel free me know if there's anything more to do on my side. :+1:

File: components/CommentReactionPicker.js [hunk 3]
```diff
@@ -96,57 +92,54 @@ const CommentReactionPicker = ({ comment, reactions }) => {
...
-                <Container
-                  border="0.1px solid #dadada"
+                <StyledCard
+                  boxShadow="-2px -1px 3px 0px #e8e8e8"
+                  zIndex={9999}
                   data-placement={placement}
                   ref={ref}
-                  style={{
-                    backgroundColor: 'white',
-                    zIndex: 9999,
```
Comment (SudharakaP) [inline components/CommentReactionPicker.js]: @Betree : Removing this zIndex; did you check whether the final comment on a conversation when you click on the emoji addition button whether the popover overlays on top of the `InlineEditField`. Notice that the [InlineEditField has a container which has zindex=2](https://github.com/opencollective/opencollective-frontend/blob/6286583a7199a78fd291c735ebcc01e5c3a9307f/components/InlineEditField.js#L176). :smiley_cat:

Comment (Betree) [reply components/CommentReactionPicker.js]: It's not removed, only moved to the component's props to take advantage of styled-components

image

Comment (SudharakaP) [reply components/CommentReactionPicker.js]: Ops, I missed it. Sorry. :+1:

File: components/conversations/CommentReactionPicker.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    const isSelected = comment.userReactions?.includes(emoji);
+    return {
+      children: <Emoji>{emoji}</Emoji>,
+      isSelected,
+      onClick: () => {
+        setOpen(false);
+        const action = isSelected ? removeCommentReaction : addCommentReaction;
+        return action({
+          variables: { emoji: emoji, comment: { id: comment.id } },
+          optimisticResponse: getOptimisticResponse(comment, emoji, !isSelected),
```
Comment (kewitz) [inline components/conversations/CommentReactionPicker.js:122]: :+1: for the optimistic response.

File: components/conversations/CommentReactionPicker.js [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+/**
+ * A component to render the reaction picker on comments.
+ */
+const CommentReactionPicker = ({ comment }) => {
+  const emojiFirstRow = ['👍️', '👎', '😀', '🎉'];
+  const emojiSecondRow = ['😕', '❤️', '🚀', '👀'];
+  const [open, setOpen] = React.useState(false);
+  const wrapperRef = React.useRef();
+  const [addCommentReaction] = useMutation(addCommentReactionMutation, mutationOptions);
+  const [removeCommentReaction] = useMutation(removeCommentReactionMutation, mutationOptions);
```
Comment (kewitz) [inline components/conversations/CommentReactionPicker.js:104]: :+1: for hooks.

**Comment by Betree** [comment]
Thanks for the review @kewitz, and thanks again for the work on this one @SudharakaP! 

Let's merge this :rocket:

**Comment by SudharakaP** [comment]
> Thanks for the review @kewitz, and thanks again for the work on this one @SudharakaP!
> 
> Let's merge this rocket

@Betree : You are welcome. Thanks for reviewing and cleaning it up. :smile:

**Pull request merged**